h2/connection.py

```diff
@@ -1634,6 +1634,33 @@ def _receive_push_promise_frame(self, frame):
 
         return frames, events + stream_events
 
+    def _handle_data_on_closed_stream(self, events, exc, frame):
+        # This stream is already closed - and yet we received a DATA frame.
+        # The received DATA frame counts towards the connection flow window.
+        # We need to manually to acknowledge the DATA frame to update the flow
+        # window of the connection. Otherwise the whole connection stalls due
+        # the inbound flow window being 0.
+        if self._stream_is_closed_by_reset(exc.stream_id):
+            frames = []
+            if frame.flow_controlled_length > 0:
+                self._inbound_flow_control_window_manager.window_opened(
```

Comment:
There is an important distinction between `window_opened()` and `process_bytes()` methods of a `WindowManager` class. I think that the right thing would be to call `process_bytes()` – to acknowledge received data, not to increase/resize the window.

Take a look into `window_opened()` method and it's docstring, there is even a possibility to get `FlowControlError` when window becomes big enough.

Reply:
Yes - you are correct!